src/ppx_odoc_of_gospel/ppx_odoc_of_gospel.ml

```diff
@@ -0,0 +1,109 @@
+open Ppxlib
+
+let is_gospel attr = attr.attr_name.txt = "gospel"
+
+let is_ocaml_doc attr =
+  attr.attr_name.txt = "ocaml.text" || attr.attr_name.txt = "ocaml.doc"
+
+(* Prefix a string with whitespace the length of gospel special comment marker *)
+let align = "   "
+let eol_and_align = "\n   "
+
+let gospel_txt_of_attributes attr =
+  let rec aux attr =
+    if is_gospel attr then
+      match attr.attr_payload with
+      | PStr
+          [
+            {
+              pstr_desc =
+                Pstr_eval
+                  ( {
+                      pexp_desc = Pexp_constant (Pconst_string (txt, loc, _));
+                      _;
+                    },
+                    attrs );
+              _;
+            };
+          ] ->
+          { txt; loc } :: (List.map aux attrs |> List.flatten)
+      | _ -> []
+    else []
+  in
+  let rec go acc = function
+    | [] -> failwith "unreachable case in go"
+    | [ { txt; loc } ] ->
+        ( align ^ String.concat eol_and_align (List.rev (txt :: acc)),
+          loc.loc_end )
+    | { txt; _ } :: xs -> go (txt :: acc) xs
+  in
```

Comment:
I would suggest replacing `go` with something along the lines of
```ocaml
align ^ String.concat eol_and_align (List.map (fun {txt;_} -> txt))
```
and
```
List.fold_left (fun _ { loc; _ } -> loc.loc_end) Lexing.dummy_pos
```
since `go` does two separate things at once.